### PR TITLE
Add setter to configure ClientOptions on processor builder

### PR DESCRIPTION
**Pull request by srnagar**
This PR adds a new setter method on the `EventProcessorClientBuilder` to configure `ClientOptions`. 

Fixes https://github.com/Azure/azure-sdk-for-java/issues/17595

### CODE REVIEW

**Review by conniey — APPROVED** [review]

**Review by JonathanGiles — COMMENTED** [review]
Where did we end with ClientOptions? This will be the protocol-agnostic API and in the future we might introduce HttpClientOptions? Does applicationId and headers both make sense as being settable here?

**Comment by srnagar** [comment]
@JonathanGiles `ClientOptions` are protocol agnostic. `HttpClientOptions` will extend from `ClientOptions` which will have HTTP-specific options. For AMQP, both application and headers are valid. So, adding this API makes sense for the processor client. Also, this API was already added to the `EventHubClientBuilder`. This PR is just adding the same API to `EventProcessorClientBuilder`.

**Comment by JonathanGiles** [comment]
Yep - as discussed, just be sure to note that the headers are applicable to the admin client rather than the amqp clients.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java [hunk 1]
```diff
@@ -211,6 +212,19 @@ public EventProcessorClientBuilder retry(AmqpRetryOptions retryOptions) {
         return this;
     }
 
+    /**
+     * Sets the client options for the processor client. The application id set on the client options will be used
```
Comment (YijunXieMS) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java:216]: I don't see the code change in this PR to use the application id from clientOptions. Is it updated in a different PR?

Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java:216]: Yeah, this is already done in core amqp.

**Pull request merged**